sklearn/feature_extraction/image.py

```diff
@@ -445,8 +445,13 @@ def extract_patches_2d(image, patch_size, *, max_patches=None, random_state=None
         return patches.reshape((n_patches, p_h, p_w))
     else:
         return patches
-
-
+@validate_params(
+    {
+        "patches" : [np.ndarray],
+        "image_size" : [tuple, list] 
+        
+    }
+)
```

Comment:
```suggestion
@validate_params({"patches" : [np.ndarray], "image_size" : [tuple, list]})
```

Reply:
I am wondering if we should use `Hidden(list)` since we don't support it officially but removing the support would be a breaking change.

This might be better.